bundles/com.eclipsesource.jaxrs.publisher/src/com/eclipsesource/jaxrs/publisher/internal/ServletContainerBridge.java

```diff
@@ -45,17 +45,19 @@ public void run() {
       try {
         Thread.currentThread().setContextClassLoader( Request.class.getClassLoader() );
         synchronized( this ) {
-          if( !isJerseyReady() ) {
-            // if jersey has not been initialized - use the init method
+          isJerseyReady = false;
+          // No WebComponent present, initialize Jersey so it's created
+          if( getServletContainer().getWebComponent() == null ) {
             getServletContainer().init( servletConfig );
-          } else {
-            // otherwise reload
-            isJerseyReady = false;
+          } 
+          // We already have a WebComponent we need to reload it
+          else {
             getServletContainer().reload( ResourceConfig.forApplication( application ) );
           }
           isJerseyReady = true;
         }
-      } catch( ServletException e ) {
+      } catch( Throwable e ) {
+        e.printStackTrace();
```

Comment:
woldn't logging be appropriate here?
In production there is rarely console output that can be read.

Reply:
I agree with you, but the fix is not from me.